src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1591 @@
+        end
+    end
+    return parameters
+end
+
+-- Modified from library
+function get_parameters(file_name, parameter_list)
+    local file_parameters = get_parameters_from_file(file_name)
+    if nil ~= file_parameters then
+        for param_name, def_val in pairs(file_parameters) do
+            local param_val = file_parameters[param_name]
+            if nil ~= param_val then
+                parameter_list[param_name] = param_val
+            end
+        end
+    end
+    return parameter_list
+end
+
+--------------------------------------
+function path_set(filename)
+    local path = finale.FCString()
+    local path_delimiter = finale.FCString()
+    local ui = finenv.UI()
+    path:SetUserOptionsPath()
+    if ui:IsOnMac() then
+        path_delimiter.LuaString = "/"
+    elseif ui:IsOnWindows() then
+        path_delimiter.LuaString = "\\"
+        --path_delimiter.LuaString = "/" -- apparently Windows can use either! Go figure!
+    end
+    path.LuaString = path.LuaString..path_delimiter.LuaString..filename
+    return path
+end
+
+function harp_config_load()
+    local path = path_set(config_filename)
+    local config_settings = {}
+    local init_settings = {root = 2, accidental = 1, scale = 0, scale_check = 1, chord = 0, chord_check = 0, diagram_check = 1, names_check = 0, partial_check = 0, stack = 1, pedal_lanes = 1, last_notes = "D, C, B, E, F, G, A"}
+    -- This next might not be needed... But doesn't hurt so leaving it in for now...
+    local init_count = 0
+    for i,k in pairs(init_settings) do
+        init_count = init_count + 1
+    end
+    --
+    local file_r = io.open(path.LuaString, "r")
+
+    if file_r == nil then
+        config_settings = init_settings
+        harp_config_save(config_settings)
+        file_r = io.open(path.LuaString, "r")
+    end
+
+    config_settings = get_parameters(path.LuaString, config_settings)
+
+    for key, val in pairs(init_settings) do
+        if config_settings[key] == nil then
+            config_settings[key] = val
+        end
+    end
+    return config_settings
+end -- harp_config_load()
+
+function harp_config_save(config_settings)
+    local path = path_set(config_filename)
+    local file_w = io.open(path.LuaString, "w")
+    for key, val in pairs(config_settings) do
+        file_w:write(key.." = "..val.."\n")
+    end
+    file_w:close()
+end -- harp_config_save()
+
+function getUsedFontName(standard_name)
+    local font_name = standard_name
+    if string.find(os.tmpname(), "/") then
+        font_name = standard_name
+    elseif string.find(os.tmpname(), "\\") then
+        font_name = string.gsub(standard_name, "%s", "")
+    end
+    return font_name
+end
+
+function get_def_mus_font()
+    local fontinfo = finale.FCFontInfo()
+    if fontinfo:LoadFontPrefs(finale.FONTPREF_MUSIC) then
+        return getUsedFontName(fontinfo:GetName())
+    end
+end
+
+---
+function harp()
+    finenv.RetainLuaState = true
+    local partial = false
```

Comment:
Recommendation: do not set `finenv.RetainLuaState = true` until after you successfully run `ShowModeless`. That's because if an error occurs while building the the dialog box, it leaves your Lua state in an unknown condition and the only way to recover is restart Finale.

Ultimately, this code will be greatly simplified by using the `RunModeless` mixin function that handles all of these details for you. I really wish we could start merging code that uses it.